components/FooterModal.tsx

```diff
@@ -0,0 +1,97 @@
+import {
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  useDisclosure,
+} from "@chakra-ui/core";
+import React, { Fragment, ReactElement, useEffect } from "react";
+import {
+  FooterLink,
+  TextStyled,
+  TextStyledBold,
+} from "@/components/StyledCore";
+import { useRouter } from "next/router";
+
+interface FooterModalProps {
+  linkBody: string;
+  modalBody: {
+    paragraph: string[];
+    title: string;
+  }[];
+}
+
+export default function FooterModal({
+  linkBody,
+  modalBody,
+}: FooterModalProps): ReactElement {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const router = useRouter();
+
+  useEffect(() => {
+    const hash = window.location.hash;
+    const thisModalHash = linkBody.replace(/\s/g, "").toLowerCase();
+    if (hash === `#${thisModalHash}`) {
+      onOpen();
+    }
+  }, [linkBody, onOpen]);
+
+  const handleOnClose = () => {
+    onClose();
+    router.replace(router.pathname);
+  };
+
+  const handleOnOpen = () => {
+    const lastCharInPath = router.pathname.charAt(router.pathname.length - 1);
+    const currentPath: string =
+      lastCharInPath === "/" ? router.pathname : `${router.pathname}/`;
+    onOpen();
+    router.replace(
+      `${currentPath}#${linkBody.replace(/\s/g, "").toLowerCase()}`
+    );
+  };
+
+  return (
+    <>
+      <FooterLink
+        _focus={{}}
+        _hover={{}}
+        fontSize={["xs", "xs", "xs", "sm"]}
+        onClick={handleOnOpen}
+        rel="noopener noreferrer"
+        target="_blank"
+      >
+        {linkBody}
+      </FooterLink>
+      <Modal
```

Comment:
Nitpick: consider adding a simple slide in animation to the modal https://chakra-ui.com/modal#adding-transition

Reply:
Will do!